src/main/java/com/commercetools/sync/sdk2/types/utils/TypeSyncUtils.java

```diff
@@ -0,0 +1,54 @@
+package com.commercetools.sync.sdk2.types.utils;
+
+import static com.commercetools.sync.sdk2.commons.utils.OptionalUtils.filterEmptyOptionals;
+import static com.commercetools.sync.sdk2.types.utils.TypeUpdateActionUtils.*;
+
+import com.commercetools.api.models.type.Type;
+import com.commercetools.api.models.type.TypeDraft;
+import com.commercetools.api.models.type.TypeUpdateAction;
+import com.commercetools.sync.sdk2.types.TypeSyncOptions;
+import java.util.List;
+import javax.annotation.Nonnull;
+
+public final class TypeSyncUtils {
+
+  /**
+   * Compares all the fields (including the field definitions see {@link
+   * TypeUpdateActionUtils#buildFieldDefinitionsUpdateActions(Type, TypeDraft, TypeSyncOptions)}) of
+   * a {@link Type} and a {@link TypeDraft}. It returns a {@link List} of {@link TypeUpdateAction}
+   * as a result. If no update actions are needed, for example in case where both the {@link Type}
+   * and the {@link TypeDraft} have the same fields, an empty {@link List} is returned.
+   *
+   * <p>Note: The commercetools API doesn't support the following:
+   *
+   * <ul>
+   *   <li>updating the inputHint of a FieldDefinition
```

Comment:
The docs says you can do it: https://docs.commercetools.com/api/projects/types#change-inputhint

Reply:
Good catch. Forgot to remove the comment. Thx.

Reply:
[4525dca](https://github.com/commercetools/commercetools-sync-java/pull/1013/commits/4525dca84db9d874c237d2eab827d8df3cd3b25e)